chain-path-derivation/src/derivation_path.rs

```diff
@@ -119,6 +120,37 @@ impl Default for DerivationPath<AnyScheme> {
     }
 }
 
+/* Comparison ************************************************************** */
+
+impl<T1, T2> PartialEq<DerivationPath<T1>> for DerivationPath<T2> {
+    fn eq(&self, other: &DerivationPath<T1>) -> bool {
+        self.path.eq(&other.path)
+    }
+}
+
+impl<T> Eq for DerivationPath<T> {}
+
+impl<T1, T2> PartialOrd<DerivationPath<T1>> for DerivationPath<T2> {
+    fn partial_cmp(&self, other: &DerivationPath<T1>) -> Option<std::cmp::Ordering> {
+        self.path.partial_cmp(&other.path)
+    }
+}
+
+impl<T> Ord for DerivationPath<T> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.path.cmp(&other.path)
+    }
+}
+
+/* Hasher ****************************************************************** */
+
+impl<T> Hash for DerivationPath<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.path.hash(state);
+        self._marker.hash(state);
```

Comment:
mostly curiosity, is it there a reason to hash `PhantomData`?

Reply:
yes, we could have a different type with the same path value. We still want to have a different hash.